### ISSUE TITLE
Cannot create wx.GraphicsContext from wx.AutoBufferedPaintDC

### ISSUE THREAD

**Issue by adrientetar** [wxWidgets/Phoenix#783]
~~~
Traceback (most recent call last):
  File "<string>", line 308, in OnPaint
    ctx = wx.GraphicsContext.Create(dc)
TypeError: GraphicsContext.Create(): arguments did not match any overloaded call:
  overload 1: argument 1 has unexpected type 'AutoBufferedPaintDC'
  overload 2: argument 1 has unexpected type 'AutoBufferedPaintDC'
  overload 3: argument 1 has unexpected type 'AutoBufferedPaintDC'
  overload 4: argument 1 has unexpected type 'AutoBufferedPaintDC'
  overload 5: argument 1 has unexpected type 'AutoBufferedPaintDC'
~~~

wxPython 4.0.1

**Comment by mesalu** [comment]
@Robind42, ideas?
I can only skim docs right now but from what I can tell either identity for `AutoBufferedPaintDC` should have a matching overload.
Either `MemoryDC` for a buffered DC, or `WindowDC` for a paint DC.

**Comment by mesalu** [comment]
Ah, I just remembered, well was reminded, that etg hacks AutoBufferedPaintDC to just look like a DC instance, so SIP cannot select an overload. (all options are further down the inheritance chain)

I'd you can avoid using AutoBufferedPaintDC, that'd be the best option. Otherwise, we're looking at options, but it may take a while to find a fix that doesn't introduce bugs.

**Comment by RobinD42** [comment]
Currently the best way to handle this would be like this:

```python
if self.IsDoubleBuffered():
    dc = wx.PaintDC(self)
else:
    dc = wx.BufferedPaintDC(self)
ctx = wx.GraphicsContext.Create(dc)
```

Or you could just always use a `wx.BufferedPaintDC` and take the tiny performance hit on the platforms where it is not needed.

**Comment by RobinD42** [comment]
Fixed by #874
